crates/bevy_ptr/src/lib.rs

```diff
@@ -8,11 +8,14 @@ use core::{
     cell::UnsafeCell, marker::PhantomData, mem::ManuallyDrop, num::NonZeroUsize, ptr::NonNull,
 };
 
-#[derive(Copy, Clone)]
 /// Used as a type argument to [`Ptr`], [`PtrMut`] and [`OwningPtr`] to specify that the pointer is aligned.
-pub struct Aligned;
 #[derive(Copy, Clone)]
+#[repr(transparent)]
```

Comment:
I am not familiar with `repr(transparent)`, but according to https://doc.rust-lang.org/nomicon/other-reprs.html#reprtransparent it should only be used on structs with a single non-zero sized field. So if I understand correctly, this should not be added to `Aligned` and `Unaligned`.

I guess the ptr types below are fine because phantom data has no size? :thinking:

Reply:
alright, As far as I took reading docs, it's unnecessary to put `repr(transparent)` into `Aligned` and `Unaligned`, which not used for FFI-boundary. Thanks for catching me up.

Reply:
@NiklasEi thought?

Reply:
Yes, I think that makes sense

Reply:
> So if I understand correctly, this should not be added to Aligned and Unaligned.

Yes, `repr(transparent)` makes no sense with ZST.

So this is wrong:

```rs
#[repr(transparent)]
struct A;
```

> I guess the ptr types below are fine because phantom data has no size?

It should be totally okay to have ZST in addition to real type, and this kind of code is all over physx-rs ([for example, here](https://github.com/EmbarkStudios/physx-rs/blob/1c6fb68824de9f9b61ac3fb2afa006e7c572ca0e/physx/src/rigid_static.rs#L24-L30)).

So this is fine:

```rs
#[repr(transparent)]
struct B<T, U> {
    data: T,
    _phantom: PhantomData<U>,
}
```